backend/rust-bindings/src/lib.rs

```diff
@@ -213,17 +178,9 @@ fn deserialize_type_value(
     }
 }
 
-fn get_str_from_pointer(input: *const c_char, result: *mut *mut c_char) -> Result<String> {
+fn get_str_from_pointer(input: *const c_char) -> Result<String> {
     let c_str: &CStr = unsafe { CStr::from_ptr(input) };
-    let str_slice: &str = match c_str.to_str() {
-        Ok(r) => r,
-        Err(e) => {
-            unsafe {
-                *result = CString::new(e.to_string()).unwrap().into_raw()
-            }
-            return Err(anyhow!("not able to parse pointer to string"))
-        },
-    };
+    let str_slice: &str = c_str.to_str()?;
```

Comment:
no need for adding context to error message?

Reply:
I just wanted to refactor, I've only moved the place the result is written to. :shrug:

Reply:
The error message before was `"not able to parse pointer to string"`. I don't know what the current msg is (might be worse, might be better) but it looks changed to me.

Reply:
```
    let schema_hex = match get_str_from_pointer(schema, result) {
        Ok(out) => out,
        Err(_) => return false,
    };
```
This was how it was used. That message was thrown away anyway. :shrug:

Reply:
Well if you have to be all whole program analysis about it...

Reply:
I think it is more clean now since you handle the error in the callee function. I see now that I could have just written the error and returned the same (which will also not be used but I avoid the custom error).